EIPS/eip-4361.md

```diff
@@ -0,0 +1,304 @@
+
+This work is sponsored by the Ethereum Foundation and Ethereum Name Service (ENS). It is being developed in the open through a series of recorded community calls and public repositories, and its development is informed by over twenty user interviews with a focus on currently-in-production uses, related prior EIPs, and fits within product roadmaps.
+
+## Specification
+Sign-In with Ethereum works as follows:
+
+1. The wallet presents the user with a structured plaintext message or equivalent interface for signing with the [EIP-191](https://eips.ethereum.org/EIPS/eip-191) signature scheme (string prefixed with `\x19Ethereum Signed Message:\n<length of message>`). The `message` MUST incorporate an Ethereum `address`, `version` of the message, `uri` for scoping, `nonce` acceptable to the server, and `issued-at` timestamp.
+2. The signature is then presented to the server, which checks the signature's validity and message content.
+3. Additional fields, including `expiration-time`, `not-before`, `request-id`, `chain-id`, and `resources` may be incorporated as part of authentication for the session.
+4. The server may further fetch data associated with the Ethereum address, such as from the Ethereum blockchain (e.g., ENS, account balances, ERC-20/ERC-721/ERC-1155 asset ownership), or other data sources that may or may not be permissioned.
+
+
+### Example message to be signed
+```
+service.org wants you to sign in with your Ethereum account:
+0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+
+I accept the ServiceOrg Terms of Service: https://service.org/tos
+
+URI: https://service.org/login
+Version: 1
+Nonce: 32891756
+Issued At: 2021-09-30T16:25:24Z
+Chain ID: 1
+Resources:
+- ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/
+- https://example.com/my-web2-claim.json
+```
+
+### Informal Message Template
+A Bash-like informal template of the full message is presented below for readability and ease of understanding. Field descriptions are provided in the following section. A full ABNF description is provided in the section thereafter.
+```
+${domain} wants you to sign in with your Ethereum account:
+${address}
+
+${statement}
+
+URI: ${uri}
+Version: ${version}
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+Not Before: ${not-before}
+Request ID: ${request-id}
+Chain ID: ${chain-id}
+Resources:
+- ${resources[0]}
+- ${resources[1]}
+...
+- ${resources[n]}
+```
+### Message Field Descriptions
+- `domain` is the [RFC 4501](https://datatracker.ietf.org/doc/html/rfc4501) `dnsauthority` that is requesting the signing.
+- `address` is the Ethereum address performing the signing conformant to capitalization encoded checksum specified in [EIP-55](https://eips.ethereum.org/EIPS/eip-55) where applicable.
+- `statement` (optional) is a human-readable ASCII assertion that the user will sign, and it must not contain `'\n'`.
```

Comment:
I might suggest also excluding `:` (colon) to avoid confusing the assertion with the subsequent metadata.

Reply:
Great suggestion. I considered this, but some implementations including OpenSea may want to include a URL to their terms of service or similar, which would require `:` to construct. Is this exception reasonable enough to you to warrant its inclusion?

Reply:
They can percent-encode it right?

Reply:
I will keep what's there for now, due to our user research. If you feel strongly about this, please continue the conversation in ETH Magicians. Thanks again.